kettle/make_json.py

```diff
@@ -103,11 +107,11 @@ def populate_path_to_job_and_number(self):
             if self.path.startswith(bucket):
                 prefix = meta['prefix']
                 break
-
+        else:
             if self.path.startswith('gs://kubernetes-jenkins/pr-logs'):
                 prefix = 'pr:'
             else:
-                raise ValueError('unknown build path')
+                raise BuildError(f'unknown build path for {self.path} in known bucket paths')
```

Comment:
The new message lgtm.  What's the rationale behind changing the type? Personally I would have kept it the same to reduce blast radius

Reply:
I have just found it more maintainable to use custom exceptions because they are usually more descriptive, if called from else where can be specifically caught from a known origin vs base exceptions that can be triggered by multiple lower level things. 

I am ok removing it to reduce the number of changes though.

Reply:
done